src/app.mm

```diff
@@ -141,19 +146,18 @@ - (void)applicationWillFinishLaunching:(NSNotification *)notification
     setenv("LC_ALL", ss.str().c_str(), 1);
     setenv("LANG", ss.str().c_str(), 1);
 
-    [self newWindow];
-
-    // Open files given on command-line
-    for (int i=1; i<g_argc; i++) {
-        [activeWindow openFilename:[NSString stringWithUTF8String:g_argv[i]]];
+    /* Check whether Neovim.app was opened via launchd or from terminal */
+    if (!(isatty (STDIN_FILENO) || isatty (STDOUT_FILENO) || isatty(STDERR_FILENO)))
+        [self newWindow];
+    else if (g_argc == 1)
+        [self newWindow];
+    else {
+        std::vector<char *> args;
+        for (int i = 1; i < g_argc ; i++)
+            args.push_back(g_argv[i]);
+
+        [self newWindowWithArgs:&args];
```

Comment:
I put the terminal check there because when you do `open build/Neovim.app` it has the arguments of `["Neovim", "-psn_X_XXXXX"]` where Xs are numbers. My initial implementation i was just checking for 2 arguments and "-psn" in the beginning of the argument second argument but I found this to be cleaner. 

Just noticed that if you start the app by clicking on it, it does not have the `-psn_X_XXXXX` argument.